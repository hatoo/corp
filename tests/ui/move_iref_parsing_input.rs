use futures::FutureExt;
use stap::{Cursor, Input, ParsingInput};

fn main() {
    let input = Input::new(Cursor {
        buf: Vec::<u8>::new(),
        index: 0,
    });

    let mut pi = ParsingInput::new(Box::new(input), |iref| async move { iref }.boxed_local());

    pi.poll();

    let iref = pi.result_mut();

    drop(pi);

    iref;
}
