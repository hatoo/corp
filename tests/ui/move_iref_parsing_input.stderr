error[E0505]: cannot move out of `pi` because it is borrowed
  --> tests/ui/move_iref_parsing_input.rs:16:10
   |
10 |     let mut pi = ParsingInput::new(Box::new(input), |iref| async move { iref }.boxed_local());
   |         ------ binding `pi` declared here
...
14 |     let iref = pi.result_mut();
   |                -- borrow of `pi` occurs here
15 |
16 |     drop(pi);
   |          ^^ move out of `pi` occurs here
17 |
18 |     iref;
   |     ---- borrow later used here
